### ISSUE TITLE
Is the leave-one-domain-out validation the same as in domainbed?

### ISSUE THREAD

**Issue by Luodian** [kakaobrain/miro#2]
Hi Dear authors of MIRO,

Thanks for providing such a wonderful work. I've noticed your training & evaluation pipeline is little different than original domainbed ones. As I understand, the training procedure put D_tr domains into train loader and leave test domain as reported OOD accuracy. Is it the same as DomainBed's leave-one-domain-out strategy or test-domain validation strategy?

```python
    # setup loaders
    train_loaders = [
        InfiniteDataLoader(
            dataset=env,
            weights=env_weights,
            batch_size=batch_size,
            num_workers=dataset.N_WORKERS,
        )
        for (env, env_weights), batch_size in iterator.train(zip(in_splits, batch_sizes))
    ]

    # setup eval loaders
    eval_loaders_kwargs = []
    for i, (env, _) in enumerate(in_splits + out_splits + test_splits):
        batchsize = hparams["test_batchsize"]
        loader_kwargs = {"dataset": env, "batch_size": batchsize, "num_workers": dataset.N_WORKERS}
        if args.prebuild_loader:
            loader_kwargs = FastDataLoader(**loader_kwargs)
        eval_loaders_kwargs.append(loader_kwargs)
```

**Comment by khanrc** [comment]
Hi, thank you for the interest in our study!

The training & evaluation pipeline is adopted from https://github.com/khanrc/swad. We used DomainBed's training-domain validation strategy for the model selection.

```py
records = Q(records)
oracle_best = records.argmax("test_out")["test_in"]
iid_best = records.argmax("train_out")["test_in"]  # training-domain validation
last = records[-1]["test_in"]

in_key = "train_out"
iid_best_indomain = records.argmax("train_out")[in_key]
last_indomain = records[-1][in_key]

ret = {
    "oracle": oracle_best,
    "iid": iid_best,
    "last": last,
    "last (inD)": last_indomain,
    "iid (inD)": iid_best_indomain,
}
```

**Comment by vellichorw** [comment]
What does {“oracle”:oracle_best} mean?

**Comment by khanrc** [comment]
@vellichorw It indicates the results of test-domain validation (oracle) model selection method strategy, not the "oracle model". It is not used in the paper.
